src/features/congregation/upcoming_events/edit_upcoming_event/useEditUpcomingEvent.tsx

```diff
@@ -1,133 +1,196 @@
-import { useState } from 'react';
+import { ChangeEvent, useState } from 'react';
 import { EditUpcomingEventProps } from './index.types';
 import { UpcomingEventCategory } from '@definition/upcoming_events';
+import { SelectChangeEvent } from '@mui/material';
+import { stackDatesToOne } from '@utils/date';
+import { addHours } from 'date-fns';
+import { useAtomValue } from 'jotai';
+import { hour24FormatState } from '@states/settings';
 
 const useEditUpcomingEvent = ({ data, onSave }: EditUpcomingEventProps) => {
-  const [localEvents, setLocalEvents] = useState(data);
+  const hour24 = useAtomValue(hour24FormatState);
```

Comment:
let's wrap all these function inside a useCallback to cache these function between rerenders
cc @rhahao